webservices/resources/sched_a.py

```diff
@@ -76,6 +80,16 @@ def args(self):
     def build_query(self, **kwargs):
         query = super().build_query(**kwargs)
         query = filters.filter_contributor_type(query, self.model.entity_type, kwargs)
+
+        if kwargs.get('contributor_zip'):
+            for value in kwargs['contributor_zip']:
+                if re.search('^\d{5}$',value) is None and re.search('^[-]\d{5}$',value) is None:
```

Comment:
Just a suggestion - `'^-?\d{5}$'` will achieve the same result with a single expression.